File: app/src/main/java/com/molihua/hlbmerge/ffmpeg/core/BaseFFmpegCallback.java
package com.molihua.hlbmerge.ffmpeg.core;

import com.blankj.molihuan.utilcode.util.FileUtils;
import com.molihua.hlbmerge.dao.ConfigData;
import com.xuexiang.xtask.XTask;
import com.xuexiang.xui.widget.dialog.materialdialog.MaterialDialog;

/**
 * @ClassName: BaseFFmpegCallback
 * @Author: molihuan
 * @Date: 2023/01/21/13:59
 * @Description:
 */
public class BaseFFmpegCallback {
    protected MaterialDialog dialog;
    protected int successNum = 0;
    protected int failNum = 0;

    //回调事件模板
    public void onFinish() {
        dialog.incrementProgress(1);
        successNum++;
        FileUtils.deleteAllInDir(ConfigData.TYPE_OUTPUT_FILE_PATH_TEMP);
    }

    public void onProgress(int progress, long progressTime) {
        XTask.postToMain(new Runnable() {
            @Override
            public void run() {
                dialog.setContent("已处理progressTime=" + (double) progressTime / 1000000 + "秒");
            }
        });
    }

    public void onCancel() {
        dialog.cancel();
    }

    public void onError(String message) {
        dialog.incrementProgress(1);
        failNum++;
    }


    public int getSuccessNum() {
        return successNum;
    }

    public int getFailNum() {
        return failNum;
    }

    public void setSuccessNum(int successNum) {
        this.successNum = successNum;
    }

    public void setFailNum(int failNum) {
        this.failNum = failNum;
    }
}

File: app/src/main/java/com/molihua/hlbmerge/ffmpeg/core/BaseFFmpegCallback.kt
package com.molihua.hlbmerge.ffmpeg.core

import com.blankj.molihuan.utilcode.util.FileUtils
import com.molihua.hlbmerge.dao.ConfigData
import com.xuexiang.xtask.XTask
import com.xuexiang.xui.widget.dialog.materialdialog.MaterialDialog

open class BaseFFmpegCallback {
    @JvmField
    protected var dialog: MaterialDialog? = null
    var successNum: Int = 0
    var failNum: Int = 0

    //回调事件模板
    fun onFinish() {
        dialog!!.incrementProgress(1)
        successNum++
        FileUtils.deleteAllInDir(ConfigData.TYPE_OUTPUT_FILE_PATH_TEMP)
    }

    fun onProgress(progress: Int, progressTime: Long) {
        XTask.postToMain(object : Runnable {
            override fun run() {
                dialog!!.setContent("进度:" + progress + "\n" + progressTime.toDouble() / 1000000)
            }
        })
    }

    fun onCancel() {
        dialog!!.cancel()
    }

    fun onError(message: String?) {
        dialog!!.incrementProgress(1)
        failNum++
    }

}